src/python/pants/backend/python/tasks2/python_binary_create.py

```diff
@@ -103,6 +103,10 @@ def _create_binary(self, binary_tgt, results_dir):
 
       builder = PEXBuilder(path=tmpdir, interpreter=interpreter, pex_info=pex_info, copy=True)
 
+      # Add binary target's interpreter compatibility to pex info
+      for constraint in binary_tgt.compatibility:
```

Comment:
I think you can scrap this: 

The closure iterated over a few lines down includes binary_tgt itself. And `has_python_sources` is true for PythonBinary targets.  So as far as I can tell binary_tgt's constraints will be added there, and there's no need to do so here.